akka-http2-support/src/main/scala/akka/http/impl/engine/http2/ResponseRendering.scala

```diff
@@ -41,4 +58,68 @@ private[http2] object ResponseRendering {
       response.entity.dataBytes
     )
   }
+
+  private def renderHeaders(
+    headersSeq:   immutable.Seq[HttpHeader],
+    headerPairs:  VectorBuilder[(String, String)],
+    serverHeader: Option[Server],
+    log:          LoggingAdapter
+  ): Unit = {
+    def suppressionWarning(h: HttpHeader, msg: String): Unit =
+      log.warning("Explicitly set HTTP header '{}' is ignored, {}", h, msg)
+
+    // optimized, as it is done for every response
+    val headers = headersSeq.toArray
```

Comment:
Yes, we might want to optimize further later on when we have some performance numbers. Right now everything is dwarfed by JVM TLS slowness anyway for Http/2.